### PR TITLE
Feature/magento2 470 add plugininfo and shopinfo

### PR DESCRIPTION
**Pull request by awesselburg**
# Pull Request Template

## Description

Add PluginInfo and ShopInfo to ping action

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: src/Model/Forwarder.php [hunk 1]
```diff
@@ -221,4 +223,40 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
 
         return $reviews;
     }
+
+    /**
+     * @return array|mixed[]
+     */
+    public function createPluginInfo()
+    {
+        $objectManager   = ObjectManager::getInstance();
```
Comment (augsteyer) [inline src/Model/Forwarder.php]: you do not use object manager here, we only use it once, please see: `$forwarderInitializer = $manager->get('Shopgate\Base\Helper\Initializer\Forwarder');`
It just instantiates the needed classes and you just use a getter to load them. The reason being is that we cannot use DI here bcs of final constructor of the extended class. However, we want to use the object manager minimally, so we only call it once here. Once in the Config (for same reason) and once in your other PR.

File: src/Model/Forwarder.php [hunk 2]
```diff
@@ -221,4 +223,40 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
 
         return $reviews;
     }
+
+    /**
+     * @return array|mixed[]
```
Comment (augsteyer) [inline src/Model/Forwarder.php]: it will just be `string[]`, right?

File: src/Model/Forwarder.php [hunk 3]
```diff
@@ -221,4 +223,40 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
...
+            'Magento-Name'    => $productMetadata->getName(),
+            'Magento-Version' => $productMetadata->getVersion(),
+            'Magento-Edition' => $productMetadata->getEdition(),
+        ];
+    }
+
+    /**
+     * get additional data from the magento instance
+     *
+     * @return array|mixed[]
```
Comment (augsteyer) [inline src/Model/Forwarder.php]: what will it be then? string[]|int[] ? array and mixed are too ambiguous, hope we can provide more clarity here as to what to expect.

File: src/Model/Forwarder.php [hunk 4]
```diff
@@ -221,4 +223,40 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
...
+
+        return [
+            'Magento-Name'    => $productMetadata->getName(),
+            'Magento-Version' => $productMetadata->getVersion(),
+            'Magento-Edition' => $productMetadata->getEdition(),
+        ];
+    }
+
+    /**
+     * get additional data from the magento instance
```
Comment (augsteyer) [inline src/Model/Forwarder.php]:
```suggestion
     * Retrieve module information from the magento instance
```

**Review by augsteyer — CHANGES_REQUESTED** [review]

File: src/Model/Forwarder.php [hunk 5]
```diff
@@ -23,6 +23,8 @@
 namespace Shopgate\Base\Model;
 
 use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Module\FullModuleList;
+use Magento\Framework\App\ProductMetadataInterface;
 use Magento\Store\Model\StoreManagerInterface;
```
Comment (andre-kraus) [inline src/Model/Forwarder.php:28]: Wrong order of use statements, just use auto formatting.

File: src/Model/Forwarder.php [hunk 6]
```diff
@@ -221,4 +230,33 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
 
         return $reviews;
     }
+
+    /**
+     * @return array|string[]
```
Comment (augsteyer) [inline src/Model/Forwarder.php]:
```suggestion
     * @return string[]
```

File: src/Model/Forwarder.php [hunk 7]
```diff
@@ -221,4 +230,33 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
...
+            'Magento-Name'    => $this->productMetadataInterface->getName(),
+            'Magento-Version' => $this->productMetadataInterface->getVersion(),
+            'Magento-Edition' => $this->productMetadataInterface->getEdition(),
+        ];
+    }
+
+    /**
+     * Retrieve module information from the magento instance
+     *
+     * @return array|string[]
```
Comment (augsteyer) [inline src/Model/Forwarder.php]:
```suggestion
     * @return string[]
```

File: src/Model/Forwarder.php [hunk 8]
```diff
@@ -221,4 +230,33 @@ protected function createReviews($limit = null, $offset = null, array $uids = []
...
+     */
+    public function createShopInfo()
+    {
+        foreach ($this->fullModuleList->getAll() as $module) {
+            $pluginResponse[] = [
+                'name'    => $module['name'],
+                'version' => $module['setup_version'],
+            ];
+        }
```
Comment (augsteyer) [inline src/Model/Forwarder.php]: I like more interesting way of doing this, try this:
```php
        return [
            'plugins_installed' => array_map(
                static function ($module) {
                    return [
                        'name'    => $module['name'],
                        'version' => $module['setup_version'],
                    ];
                },
                $this->fullModuleList->getAll()
            )
        ];
```

**Review by augsteyer — CHANGES_REQUESTED** [review]

**Review by augsteyer — APPROVED** [review]

File: src/Helper/Initializer/Forwarder.php [hunk 9]
```diff
@@ -22,6 +22,8 @@
 
 namespace Shopgate\Base\Helper\Initializer;
 
+use Magento\Framework\Module\FullModuleList;
+use Magento\Framework\App\ProductMetadataInterface;
```
Comment (andre-kraus) [inline src/Helper/Initializer/Forwarder.php:25]: Sort order is off

**Review by andre-kraus — CHANGES_REQUESTED** [review]

**Review by andre-kraus — APPROVED** [review]

**Pull request merged**